### DISCUSSION TITLE
Track if submit has failed

### DISCUSSION THREAD

**Discussion by santino** [react-hook-form/react-hook-form#4128] [Q&A]
I want to know if the form has been submitted and the last submit attempt has failed; not necessarily from the server, but even locally if for instance validation fails when the user attempts to submit the form.

I see `formState` exposes `isSubmitted` and `isSubmitSuccessful`, however they don't seem to quite fit my use case.
`isSubmitted` turns `true` after submitting, but it then becomes false as soon as any event happens on the form; even a blur. I believe this is probably because the form gets reset?
`isSubmitSuccessful` is `false` by default; so I cannot actually use it to tell me if a form has been submitted or not.

Am I missing something or doing anything wrong?

**Comment by bluebill1049** [comment]
what do you mean by failed? you may the request you send inside `handleSubmit`? or the form validation failed.

**Reply by santino** [reply]
By "submit", I mean when the user clicks on submit button.
From a user's point of view, when they click on the submit button, they are submitting; what happens after that is not necessarily relevant for what I want to achieve.
The form may actually never be sent to the server because there are validation errors, or there aren't validation errors and it is sent to the server and the server rejects the request may be because of other errors, or a server failure, or whatever else.

My question is: if the user clicks on submit button and the form contains errors (which it will even if they are only "local" validation errors that prevent the form even reaching the server); is there any metadata that tells me that I have an actual attempt to send the form that wasn't successful? And can I rely on this metadata until the user submits again; rather than having this reset as soon as any event is triggered on the form?

Something like `submitFailed` could do. Where, as described above, `isSubmitSuccessful` is not reliable because has always a value even if the user didn't actually attempt to submit the form; and `isSubmitted` gets reset.

**Reply by bluebill1049** [reply]
`submitCount` ?

**Reply by santino** [reply]
I think that's right, this should do:
`const submitFailed = Boolean(submitCount && !isSubmitSuccessful)`

Somehow, I guess, I didn't see things clearly until I had to write down my statement for this ticket.

BTW I have to compose this logic in a few places; so if you feel like adding this meta to `formState` is a good use-case; that would be appreciated 😉
Sounds like something simple and inexpensive.
